### PR TITLE
Chore/omega 100 clean up routing

### PR DESCRIPTION
**Pull request by eTallang**
OMEGA-100

## What is the reason for this change?

Previously, routing on min side was done through a combination of entries in `basePaths.json` and hard coded strings.

### What is the expected result?

This change removes all hard coded strings and introduces a single source for all routes.

## How is it solved?

A new file `routing.ts` is introduced that contains all routes and a few utility functions to create and manipulate urls.

### CODE REVIEW

File: packages/web/src/routing.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const urlParts = url.split('/')
+
+  if (!urlParts.length) {
+    return ''
+  }
+
+  const lastPart = urlParts.pop()!
+
+  if (lastPart.startsWith(':')) {
+    if (url === routes.meteringPoint);
```
Comment (steinmy) [inline packages/web/src/routing.ts:51]: What is happening here? It seems like this line does nothing, or perhaps I'm misunderstanding?

**Review by steinmy — CHANGES_REQUESTED** [review]

File: packages/web/src/routing.ts (hunk 1 above)
Comment (t0ff3r) [reply packages/web/src/routing.ts:51]: I agree

**Review by t0ff3r — APPROVED** [review]
Think this looks great! Maybe remove that if doing nothing and consider if /faktura should be /fakturaer for consistency in single/plural. Other than that I'd say ship it :-)

File: packages/web/src/routing.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  faq: 'sporsmal-og-svar',
+  myPage: '/minside',
+  customerPage: `/minside/:${customerIdToken}`,
+  meteringPoints: `/minside/:${customerIdToken}/malere`,
+  meteringPoint: `/minside/:${customerIdToken}/malere/:${contractIdToken}`,
+  customerInfo: `/minside/:${customerIdToken}/kundeinfo`,
+  accessOverview: `/minside/:${customerIdToken}/tilganger`,
+  hanPort: `/minside/:${customerIdToken}/hanport/:${contractIdToken}`,
+  hanPortSuccess: `/minside/:${customerIdToken}/hanport/success`,
+  invoices: `/minside/:${customerIdToken}/faktura/:${contractIdToken}`,
```
Comment (t0ff3r) [inline packages/web/src/routing.ts:35]: /faktura could maybe be fakturaer since we use tilganger/malere

File: packages/web/src/routing.ts (hunk 1 above)
Comment (eTallang) [reply packages/web/src/routing.ts:51]: Good catch. Strange that it ended up in the PR. Removed.

**Comment by marasva** [comment]
If I try to log in with Randi Rystad or Ali Jensen in test and my own account in prod I get this error message: 
`Error: Number of provided params to getUrlWithParams does not match with required url params.`

File: packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx [hunk 3]
```diff
@@ -27,7 +28,7 @@ export const CustomerRelationOverview: React.FC = () => {
 
   useEffect(() => {
     if (!isLoading && customerIds.length === 1) {
-      navigate(`/minside/kundeforhold/${customerIds[0]}`)
+      navigate(getUrlWithParams(routes.myPage, customerIds[0]))
```
Comment (marasva) [inline packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:31]: In relation to my previous comment it worked for me by just changing routes.myPage to routes.customerPage here.

Comment (eTallang) [reply packages/web/src/templates/CustomerRelationOverview/CustomerRelationOverview.tsx:31]: A very good catch.

**Pull request merged**